comp/core/agenttelemetry/impl/agenttelemetry.go

```diff
@@ -121,13 +127,14 @@ func createAtel(
 	}
 
 	return &atel{
-		enabled: true,
-		cfgComp: cfgComp,
-		logComp: logComp,
-		telComp: telComp,
-		sender:  sender,
-		runner:  runner,
-		atelCfg: atelCfg,
+		enabled:              true,
+		cfgComp:              cfgComp,
+		logComp:              logComp,
+		telComp:              telComp,
+		sender:               sender,
+		runner:               runner,
+		atelCfg:              atelCfg,
+		prevPromMetricValues: make(map[string]interface{}),
```

Comment:
Would it make more sense to have two separate maps? One for counters and one for histograms in which has concrete types. 

```go
prevPromHistogramValues: make(map[string] uint64),
prevPromCounterValues: make(map[string]float64),
```

That way we avoid the value cast 😄

Reply:
I contemplated it, originally decided against it because I have a bigger single function code and multiple maps would make it even bigger. But now after refactoring I do not see problem with that, will do.